app/src/pages/inside/projectSettingsPageContainer/projectSettingsPageContainer.jsx

```diff
@@ -66,20 +67,28 @@ export const ProjectSettingsPageContainer = () => {
   const [headerNodes, setHeaderNodes] = useState({});
 
   const createTabLink = useCallback(
-    (tabName, extendedParams = {}) => ({
-      type: PROJECT_SETTINGS_TAB_PAGE,
+    (tabName, extendedParams = {}, page = PROJECT_SETTINGS_TAB_PAGE) => ({
+      type: page,
       payload: { projectId, settingsTab: tabName, ...extendedParams },
     }),
     [projectId],
   );
 
   const extensionsConfig = useMemo(() => {
-    return extensions.reduce(
-      (acc, extension) => ({
+    return extensions.reduce((acc, extension) => {
+      const hasInitialPage = extension.initialPage?.payload || extension.initialPage?.type;
+
+      return {
         ...acc,
         [extension.name]: {
           name: extension.title || extension.name,
-          link: createTabLink(extension.name),
+          link: hasInitialPage
+            ? createTabLink(
+                extension.name,
+                { subTab: extension.initialPage?.payload?.subTab },
+                extension.initialPage?.type,
+              )
+            : createTabLink(extension.name),
```

Comment:
Can we simplify this?
In case the `initialPage?.payload` or `initialPage?.type` will be empty while passing them to the `createTabLink` function, the default values will be used.
So,
```suggestion
          link: createTabLink(extension.name, extension.initialPage?.payload, extension.initialPage?.type),
```

Reply:
Good point! This hasInitialPage function was added to explicitly check the data for readability, but you are right, the code looks overcomplicated with it, so I removed it.